Automattic/jetpack
# Properly Gate Newsletter Subscriptions

Fixes https://github.com/Automattic/gold/issues/64

## Proposed changes:

Properly Gate Newsletter Subscriptions. Currently, any user with any paid membership plan can access a "Paid Subscribers" newsletter gated post.

The fix can be found below. Note that in the last seconds of the video, I refreshed the browser several times to verify that I wouldn't get access to a Paid Newsletter gated post after subscribing to a premium content plan.

https://github.com/Automattic/jetpack/assets/7076981/6a570ba4-f92f-4858-9aaa-462d417e9c91


### Other information:


## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Subscribe via the premium content block in incognito
* Open a paid subscriber newsletter gated post in the same browser but a different tab to reuse the saved cookie
* Verify that subscribing to premium content doesn't give access to the newsletter-gated post by refreshing the newsletter-gated post.

**Note:** In the past, users could choose a newsletter paid plan as a paid plan option for premium content.  If your premium content block currently uses a newsletter-paid plan as an option, I suggest changing the plan to a non-newsletter plan before testing this PR.
